proto/c1/connector/v2/resource.proto

```diff
@@ -88,6 +89,123 @@ service ResourcesService {
   rpc ListResources(ResourcesServiceListResourcesRequest) returns (ResourcesServiceListResourcesResponse);
 }
 
+service ResourceManagerService {
+  rpc CreateResource(CreateResourceRequest) returns (CreateResourceResponse);
+  rpc DeleteResource(DeleteResourceRequest) returns (DeleteResourceResponse);
+}
+
+service CredentialManagerService {
+  rpc RotateCredential(RotateCredentialRequest) returns (RotateCredentialResponse);
+}
+
+service AccountManagerService {
+  rpc CreateAccount(CreateAccountRequest) returns (CreateAccountResponse);
+}
+
+message CreateResourceRequest {
+  Resource resource = 1;
+}
+
+message CreateResourceResponse {
+  Resource created = 1;
+  repeated google.protobuf.Any annotations = 2;
+}
+
+message DeleteResourceRequest {
+  ResourceId resource_id = 1;
+}
+
+message DeleteResourceResponse {
+  repeated google.protobuf.Any annotations = 1;
+}
+
+message RotateCredentialRequest {
```

Comment:
Are credentials always tied to a single resource? Which resource do I include here?

Reply:
I'm curious if credentials should be able to be tied to grants?

Reply:
Yes, credentials are tied to a single resource like a user. Technically you can include any resource and it's up to the connector to handle it or return an error. Postgres is kind of weird in that users are actually roles, and any role can be given credentials (which basically makes it a user). So you can technically send a rotatecredentials request with a role id, and postgres will `ALTER ROLE rolename WITH PASSWORD ...`. This also lets us do things like rotate API keys on some non-user resource.

After delving into the ticket processing code in c1, I think it's simpler to tie credentials to resources (accounts) instead of grants. A grant can expire or be revoked, but the credential for the relevant account could still be valid. Also credentials on resources is how most applications model them. If a user loses their password they can always do a rotate credential request.